basics/signals/src/index.ts

```diff
@@ -28,62 +26,54 @@ namespace CommandIDs {
 const extension: JupyterFrontEndPlugin<void> = {
   id: 'signals',
   autoStart: true,
-  requires: [ICommandPalette, ILauncher, IMainMenu],
+  optional: [ILauncher],
+  requires: [ICommandPalette, IMainMenu],
   activate: activate
 };
 
 function activate(
   app: JupyterFrontEnd,
   palette: ICommandPalette,
-  launcher: ILauncher,
-  mainMenu: IMainMenu
+  mainMenu: IMainMenu,
+  launcher: ILauncher | null
 ) {
   const manager = app.serviceManager;
   const { commands, shell } = app;
-  let category = 'Tutorial';
+  let category = 'Examples';
 
   // Add launcher
-  launcher.add({
-    command: CommandIDs.create,
-    category: category
-  });
+  if (launcher) {
+    launcher.add({
+      command: CommandIDs.create,
+      category: category
+    });
+  }
 
   function createPanel() {
-    let panel: TutorialPanel;
+    let panel: ExamplePanel;
     return manager.ready.then(() => {
-      panel = new TutorialPanel();
+      panel = new ExamplePanel();
       shell.add(panel, 'main');
       return panel;
     });
   }
 
-  // add menu tab
-  let tutorialMenu: Menu = new Menu({ commands });
-  tutorialMenu.title.label = 'Tutorial';
-  mainMenu.addMenu(tutorialMenu);
+  // Add menu tab
+  let exampleMenu: Menu = new Menu({ commands });
+  exampleMenu.title.label = 'Examples';
+  mainMenu.addMenu(exampleMenu);
 
-  // add commands to registry
+  // Add commands to registry
   let command = CommandIDs.create;
   commands.addCommand(command, {
-    label: 'Ex5: open Panel',
-    caption: 'Open the Labtutorial Extension',
+    label: 'Signal example: open Panel',
```

Comment:
Should we use `CommandIDs.create` directly in the call to to `addCommand`?

Reply:
yes, you are right. Let's not create to many intermediate variables.